src/source/Ice/TurnConnection.c

```diff
@@ -1226,137 +988,100 @@ BOOL turnConnectionGetRelayAddress(PTurnConnection pTurnConnection, PKvsIpAddres
     return FALSE;
 }
 
-STATUS turnConnectionTimerCallback(UINT32 timerId, UINT64 currentTime, UINT64 customData)
+STATUS checkTurnPeerConnections(PTurnConnection pTurnConnection)
 {
-    UNUSED_PARAM(timerId);
-    UNUSED_PARAM(currentTime);
     STATUS retStatus = STATUS_SUCCESS, sendStatus = STATUS_SUCCESS;
-    PTurnConnection pTurnConnection = (PTurnConnection) customData;
-    BOOL locked = FALSE, stopScheduling = FALSE;
     PTurnPeer pTurnPeer = NULL;
     PStunAttributeAddress pStunAttributeAddress = NULL;
     PStunAttributeChannelNumber pStunAttributeChannelNumber = NULL;
-    PStunAttributeLifetime pStunAttributeLifetime = NULL;
     UINT32 i = 0;
 
+    // turn mutex is assumed to be locked.
```

Comment:
I see this comment, but we do not end up locking in the places where this API is being used in the state machine. Am I missing where the locking is happening?

Reply:
The lock is occurring above the state machine APIs, in step state machine.

Reply:
Ok I see it. Thanks.